pliers/extractors/text.py

```diff
@@ -453,98 +446,282 @@ def __init__(self,
+            out = pool_func(preds[0][:, 1:-1, :], axis=1, keepdims=True)
+        else:
+            out = preds[1]
+        data = [out.tolist()]
+        feat = ['encoding']
+        if self.return_metadata:
+            data += [tok]
+            feat += ['sequence']   
+        return data, feat, ons, dur
+    
+    def _get_model_attributes(self):
+        return ['pretrained_model', 'framework', 'model_class', 
+                'pooling', 'return_sep', 'tokenizer_type']
 
-        data = [encodings.tolist(), encoded_tokens, t_text]
-        features = ['encoding', 'token', 'word']
 
-        return ExtractorResult(data, stims, self,
-                               features=features,
-                               onsets=t_ons, durations=t_dur)
+class BertLMExtractor(BertExtractor):
 
-    def _to_df(self, result, model_attributes=True):
-        df_dict = dict(zip(result.features, result._data))
+    ''' Use BERT for masked words prediction.
 
-        if model_attributes:
-            log_dict = {attr: getattr(result.extractor, attr) for
-                        attr in ['pretrained_model', 'encoding_level',
-                        'pooling', 'framework', 'tokenizer_type']}
-            df_dict.update(log_dict)
+    Args:
+        pretrained_model (str): A string specifying which BERT
+            model to use. Can be one of pretrained BERT models listed at
+            https://huggingface.co/transformers/pretrained_models.html
+            (valid values include all the models with 'bert' prefix) 
+            or path to custom model.
+        tokenizer (str): Type of tokenization used in the tokenization step.
+            If different from model, out-of-vocabulary tokens may be treated as
+            unknown tokens.
+        framework (str): name deep learning framework to use. Must be 'pt'
+            (PyTorch) or 'tf' (tensorflow). Defaults to 'pt'.
+        mask (int or str): Words to be masked (string) or indices of 
+            words in the sequence to be masked (indexing starts at 0). Can
+            be either a single word/index or a list of words/indices.
+            If str is passed and more than one word in the input matches 
+            the string, only the first one is masked. 
+        top_n (int): Specifies how many of the highest-probability tokens are
+            to be returned. Mutually exclusive with target and threshold.
+        target (str or list): Vocabulary token(s) for which probability is to 
+            be returned. Tokens defined in the vocabulary change across 
+            tokenizers. Mutually exclusive with top_n and threshold.
+        threshold (float): If defined, only values above this threshold will
+            be returned. Mutually exclusive with top_n and target.
+        return_softmax (bool): if True, returns probability scores instead of 
+            raw predictions scores.
+        return_metadata (bool): if True, returns masked word (if defined in the
```

Comment:
`return_metadata` seems like too broad a name for this specific piece of information (and perhaps inconsistent with the way `metadata` is used in the other classes). Maybe something like `include_masked_word`?

Reply:
now turned into `returned_masked_word` for LM extractor. Similar arguments are `return_sequence` for `BertSequenceExtractor`, and `return_token` for `BertExtractor`.